websites/S/Sorryfield/presence.ts

```diff
@@ -0,0 +1,178 @@
+	CHART_INTERMEDIATE = "https://i.imgur.com/hRMaDwD.jpg",
+	CHART_EASY = "https://i.imgur.com/BLQqUN1.jpg",
+	CHART_NEW = "https://i.imgur.com/PpA5uhv.jpg",
+	CHART_HISTORY = "https://i.imgur.com/LxwfWhY.jpg",
+	CHART_POSSESSION = "https://i.imgur.com/C42Fbsi.jpg",
+	CHART_HOT = "https://i.imgur.com/xt6IZxO.jpg",
+	CHART_COLD = "https://i.imgur.com/E9xnBoS.jpg",
+	CHART_MULTIPLAYER = "https://i.imgur.com/fdGfvt2.jpg",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			details: "Sorryfield",
+			smallImageKey: Assets.LOGO,
+		},
+		{ pathname, href } = document.location;
+	if (
+		pathname === "/" ||
+		pathname === "/sorrygle" ||
+		pathname.startsWith("/song/")
+	) {
+		presenceData.largeImageKey = Assets.LOGO;
+		delete presenceData.smallImageKey;
+	}
+	if (pathname === "/") presenceData.details = "곡 선택 중";
+	if (pathname === "/sorrygle") presenceData.details = "쏘리글";
+	if (pathname.startsWith("/song/")) {
+		presenceData.details = "노래 듣는 중";
+		let songArtist = document
+			.querySelector(".song")
+			.children[2].innerHTML.split("&nbsp;")[0];
+		/* Why innerHTML?
+		 * Because removing another expression of artist.
+		 * Other expressions of artist name and song name are separated by '&nbsp;' in Sorryfield.
+		 * '.textContent' does not have '&nbsp;'.
+		 */
+		if (document.querySelector(".title").textContent.includes("fa-lock"))
+			songArtist = `🔒 ${songArtist}`; // Hidden song from the search.
+		presenceData.state = `${songArtist} - ${document
+			.querySelector(".title")
+			.textContent.trim()}`;
+		presenceData.buttons = [{ label: "Listen", url: href }];
+	}
+	if (pathname.startsWith("/java")) {
+		presenceData.details = "자바!";
+		presenceData.largeImageKey = Assets.LOGO;
+		delete presenceData.smallImageKey;
+		if (pathname.includes("/edit")) {
+			presenceData.state = `채보 수정 중 - ${document
+				.querySelector("title")
+				.textContent.replace(" - 자바! - 쏘리들", "")}`;
+		} else if (pathname.includes("/studio")) {
+			presenceData.state = document
+				.querySelector("title")
+				.textContent.replace(" - 자바! - 쏘리들", "");
+		} else if (pathname.includes("/shop")) presenceData.state = "상점";
+		else if (pathname.includes("/sync")) presenceData.state = "환경 설정";
+		else if (href.includes("/java/")) {
+			presenceData.state = document
+				.querySelector(".title")
+				.children[1].firstChild.textContent.trim();
+		} else if (href.includes("?")) {
+			presenceData.details = "자바! 싱글플레이어";
+			const chartHeader = document.querySelector(".chart-header");
+			presenceData.state = `${
+				chartHeader.children[2].children[0].innerHTML
+			} Lv.${document.querySelector(".level").innerHTML} / ${
+				chartHeader.children[1].children[1].innerHTML
+			} - ${chartHeader.children[1].children[0].innerHTML}`;
+			presenceData.buttons = [{ label: "채보 플레이하기", url: href }];
+		} else {
+			const image = document
+				.querySelector(".active>div>img")
+				.attributes[1].value.replace("/media/images/chart-", "")
```

Comment:
This should be a bit clearer.
```suggestion
				.querySelector<HTMLImageElement>(".active>div>img")
				.src.replace("/media/images/chart-", "")
```